scalahost/nsc/src/main/scala/scala/meta/internal/databases/AnchorOps.scala

```diff
@@ -0,0 +1,30 @@
+package scala.meta.internal
+package scalahost
+package databases
+
+import java.io.{File => JFile}
+import scala.{meta => m}
+import scala.reflect.internal.util.{Position => GPosition, SourceFile => GSourceFile}
+import scala.reflect.io.{AbstractFile => GFile, PlainFile => GPlainFile}
+import scala.meta.internal.io.PlatformIO
+
+trait AnchorOps { self: DatabaseOps =>
+
+  implicit class XtensionGSourceFileLocation(gsource: GSourceFile) {
+    def toAbsolutePath: m.AbsolutePath = gsource.file.toAbsolutePath
+  }
+
+  implicit class XtensionGFileAddr(gfile: GFile) {
+    def toAbsolutePath: m.AbsolutePath = gfile match {
+      case gfile: GPlainFile => m.AbsolutePath(gfile.file)
+      case other => sys.error(s"unsupported file " + other)
+    }
+  }
+
+  implicit class XtensionGPositionAnchor(pos: GPosition) {
+    def toAnchor: m.Anchor = {
+      assert(pos.isRange)
```

Comment:
This line is problematic, I hit on an error (with stacktrace https://gist.github.com/olafurpg/aa8a42c186bce4e51d79e637420b60bf) which I suspect was caused by a for-comprehension. The culprit code looked something like 
```scala
    val patches = for {
      s <- mirror.database.sources
      (Anchor(_, start, _), sugar) <- s.sugars
      if !sugar.startsWith("[")
      patch <- tokensByOffset.get(start).toList.map {
        case tok @ Trivia() =>
          ctx.addLeft(tok, sugar.trimSugar)
        case tok =>
          ctx.addRight(tok, sugar.trimSugar)
      }
    } yield patch
```